### PR TITLE
Directly wire ctx into http request

### PR DESCRIPTION
**Pull request by rht**
Updated version of #1991.

- [x] put ctxhttp into  Godep. Blocked on `godep update` ("no buildable Go source files in $GOPATH/src/github.com/StackExchange/wmi")

### CODE REVIEW

**Comment by lgierth** [comment]
> Blocked on godep update ("no buildable Go source files in $GOPATH/src/github.com/StackExchange/wmi")

Yeah I had some fun with this one as well. It looks like the revision we specify doesn't exist in the repo.

**Comment by cryptix** [comment]
I stumbled over these was well before. Either the remote origin force-pushed their history away or somebody vendored an unpublished commit.

In any case, you manually need to rip out the code from `_workspace` and the `godeps.json`...

**Comment by whyrusleeping** [comment]
ah yeah, this might be my fault partially. The code in that repo is set to only build on windows, and since godeps only respects your current build tags, it would never vendor it (or keep it vendored). I had to add a `wmi_unix.go` to make things vendor properly.

**Comment by whyrusleeping** [comment]
tldr; i hate godeps

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: ff76771cc6edf547e9af83221f041e95d1cc430f
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by: .* <.*>**

[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) and [a script](https://github.com/ipfs/community/blob/master/dev/tools/hooks/setup_commit_msg_hook.sh) are available to help. Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23).

---
This message was auto-generated by https://gitcop.com

**Comment by rht** [comment]
> tldr; i hate godeps

From the cause, it is because godeps doesn't anticipate the windows-only case.
(though `godep update -d` like `go get -u -d` would have solved the problem)

**Comment by whyrusleeping** [comment]
I think this change is unnecessary as we can just do:
```
httpReq.Cancel = req.Context.Done()
```

for the same effect

**Comment by rht** [comment]
That's true, and that `ctxhttp.Do` runs an extra goroutine.
